src/Core/Services/AuthService.cs

```diff
@@ -268,12 +268,9 @@ private async Task<SymmetricCryptoKey> MakePreloginKeyAsync(string masterPasswor
                     kdfIterations = preloginResponse.KdfIterations;
                 }
             }
-            catch (ApiException e)
+            catch (ApiException e) when (e.Error == null || e.Error.StatusCode != System.Net.HttpStatusCode.NotFound)
             {
-                if (e.Error == null || e.Error.StatusCode != System.Net.HttpStatusCode.NotFound)
-                {
-                    throw e;
-                }
+                throw;
```

Comment:
If I'm not mistaken what is written there is not the same as before:
```c#
catch (ApiException e)
{
      if (e.Error == null || e.Error.StatusCode != System.Net.HttpStatusCode.NotFound)
      {
             throw e;
      }
}
```
Because on this code (before), all `ApiException` are being catched but only a few rethrow the exception (the others are ignored). On yours, it'll only catch the `ApiException` that comply with the condition and others will be rethrown as well (so all `ApiException` will thrown). E.g., `ApiException` with `StatusCode` equal to `NotFound` on the original code it'll be ignored grabbed by the catch, and on your code it won't be catched so the exception will be bubbled up to the caller

Reply:
I was a little confused by this as well. In another file we used the condition:
```c#
if (e.Error != null && e.Error.StatusCode == System.Net.HttpStatusCode.NotFound)
{
    return new List<BreachAccountResponse>();
}
throw e;
```
which makes more sense to me. 

Either way I've reverted this to catch all `ApiException`. I'm not quite sure what the expected behavior is either, but for now we are following the same rules as before.